sopel/modules/url.py

```diff
@@ -165,6 +185,26 @@ def process_urls(bot, trigger, urls):
     """
     shorten_url_length = bot.config.url.shorten_url_length
     for url in urls:
+        parsed = urlparse(url)
+
+        # Prevent private addresses form being queried if enable_private_resolution is False
+        if not bot.config.url.enable_private_resolution:
```

Comment:
I think this check can be performed *after* the exclusion char check: if the URL is excluded, I don't think we need to check if it's private, as it'll be discarded right away.

Reply:
... and probably *after* the `check_callback` call too.

Reply:
Moved the code after check_callback